### ISSUE TITLE
Intel OneAPI Toolkit Release ohpc

### ISSUE THREAD

**Issue by ruralsysadmin** [openhpc/ohpc#1901]
Following the install guide for v3.0 to install intel oneapi toolkit (chapter 4.7), step 3 to install the actual compilers eventually fails with the following:

GPG key at https://yum.repos.intel.com/intel-gpg-keys/GPG-PUB-KEY-INTEL-SW-PRODUCTS.PUB (0x53D04109) is already installed
The GPG keys listed for the "Intel(R) oneAPI repository" repository are already installed but they are not correct for this package.

GPG Keys are configured as: https://yum.repos.intel.com/intel-gpg-keys/GPG-PUB-KEY-INTEL-SW-PRODUCTS.PUB
Public key for intel-oneapi-compiler-fortran-common-2023.1.0-2023.1.0-46305.noarch.rpm is not installed. Failing package is: intel-oneapi-compiler-fortran-common-2023.1.0-2023.1.0-46305.noarch

If I don't use the *-ohpc package definition and install a oneapi package/toolkit directly the install goes fine. I was trying to use the ohpc method since I am having issues with the modulesetup that intel ships, wanted to see if following the OpenHPC recipe would yield a different result before digging deeper

**Comment by adrianreber** [comment]
> If I don't use the *-ohpc package definition and install a oneapi package/toolkit directly the install goes fine.

Based on which documentation do you do that?

Just that we know what you are using.

**Comment by ruralsysadmin** [comment]
> > If I don't use the *-ohpc package definition and install a oneapi package/toolkit directly the install goes fine.
> 
> Based on which documentation do you do that?
> 
> Just that we know what you are using.

If I follow the Intel docs for OneAPI the install goes fine. Create the Repo then dnf install intel-basekit
 
https://www.intel.com/content/www/us/en/developer/tools/oneapi/base-toolkit-download.html?operatingsystem=linux&distributions=dnf

**Comment by adrianreber** [comment]
Hmm, looking at the documentation I see:
```shell
[sms]# yum -y install intel-oneapi-toolkit-release-ohpc
[sms]# rpm --import https://yum.repos.intel.com/intel-gpg-keys/GPG-PUB-KEY-INTEL-SW-PRODUCTS.PUB
[sms]# yum -y install intel-compilers-devel-ohpc
[sms]# yum -y install intel-mpi-devel-ohpc
```

Installing `intel-oneapi-toolkit-release-ohpc` gives the following repository:
```shell
# cat oneAPI.repo 
[oneAPI]
name=Intel(R) oneAPI repository
baseurl=https://yum.repos.intel.com/oneapi
enabled=1
gpgcheck=1
repo_gpgcheck=1
gpgkey=https://yum.repos.intel.com/intel-gpg-keys/GPG-PUB-KEY-INTEL-SW-PRODUCTS.PUB
```
Which is exactly the same as the one from the link you mentioned from Intel.

I get following error:
```
Intel® oneAPI repository                                                                                                                                                                                   6.7 kB/s | 943  B     00:00    
Importing GPG key 0x53D04109:
 Userid     : "CN=Intel(R) Software Development Products"
 Fingerprint: E9BF 0AFC 46D6 E8B7 DA58 82F1 BAC6 F0C3 53D0 4109
 From       : https://yum.repos.intel.com/intel-gpg-keys/GPG-PUB-KEY-INTEL-SW-PRODUCTS.PUB
Key imported successfully
Public key for intel-oneapi-compiler-fortran-2023.1.0-2023.1.0-46305.x86_64.rpm is not installed. Failing package is: intel-oneapi-compiler-fortran-2023.1.0-2023.1.0-46305.x86_64
 GPG Keys are configured as: https://yum.repos.intel.com/intel-gpg-keys/GPG-PUB-KEY-INTEL-SW-PRODUCTS.PUB
Public key for intel-oneapi-compiler-fortran-common-2023.1.0-2023.1.0-46305.noarch.rpm is not installed. Failing package is: intel-oneapi-compiler-fortran-common-2023.1.0-2023.1.0-46305.noarch
 GPG Keys are configured as: https://yum.repos.intel.com/intel-gpg-keys/GPG-PUB-KEY-INTEL-SW-PRODUCTS.PUB
Public key for intel-oneapi-compiler-fortran-runtime-2023.1.0-2023.1.0-46305.x86_64.rpm is not installed. Failing package is: intel-oneapi-compiler-fortran-runtime-2023.1.0-2023.1.0-46305.x86_64
 GPG Keys are configured as: https://yum.repos.intel.com/intel-gpg-keys/GPG-PUB-KEY-INTEL-SW-PRODUCTS.PUB
Public key for intel-oneapi-mkl-2023.1.0-2023.1.0-46342.x86_64.rpm is not installed. Failing package is: intel-oneapi-mkl-2023.1.0-2023.1.0-46342.x86_64
 GPG Keys are configured as: https://yum.repos.intel.com/intel-gpg-keys/GPG-PUB-KEY-INTEL-SW-PRODUCTS.PUB
Public key for intel-oneapi-mkl-common-2023.1.0-2023.1.0-46342.noarch.rpm is not installed. Failing package is: intel-oneapi-mkl-common-2023.1.0-2023.1.0-46342.noarch
 GPG Keys are configured as: https://yum.repos.intel.com/intel-gpg-keys/GPG-PUB-KEY-INTEL-SW-PRODUCTS.PUB
Public key for intel-oneapi-mkl-common-devel-2023.1.0-2023.1.0-46342.noarch.rpm is not installed. Failing package is: intel-oneapi-mkl-common-devel-2023.1.0-2023.1.0-46342.noarch
 GPG Keys are configured as: https://yum.repos.intel.com/intel-gpg-keys/GPG-PUB-KEY-INTEL-SW-PRODUCTS.PUB
Public key for intel-oneapi-mkl-devel-2023.1.0-2023.1.0-46342.x86_64.rpm is not installed. Failing package is: intel-oneapi-mkl-devel-2023.1.0-2023.1.0-46342.x86_64
 GPG Keys are configured as: https://yum.repos.intel.com/intel-gpg-keys/GPG-PUB-KEY-INTEL-SW-PRODUCTS.PUB
Public key for intel-oneapi-mpi-2021.9.0-2021.9.0-43482.x86_64.rpm is not installed. Failing package is: intel-oneapi-mpi-2021.9.0-2021.9.0-43482.x86_64
 GPG Keys are configured as: https://yum.repos.intel.com/intel-gpg-keys/GPG-PUB-KEY-INTEL-SW-PRODUCTS.PUB
```

Which is unfortunately not our fault but Intel's fault. OpenHPC has been built against an older version of the Intel compiler and these older versions are signed with an older GPG key. Intel recently replaced the expired key with a newer key but didn't resign the old package with the new key. I think Intel needs to sign the  packages with the correct key.

@jcsiadal anything you can do to help to get old packages signed with the current key?

**Comment by jcsiadal** [comment]
Looking into this. I'll contact someone today about the issue.

**Comment by ruralsysadmin** [comment]
As of this message I can install intel-compilers-devel-ohpc however when attempting  intel-mpi-devel-ohpc I get the following error:

Running transaction
  Preparing        :                                                                                                                    1/1
  Running scriptlet: intel-mpi-devel-ohpc-2023.1-300.ohpc.7.1.x86_64                                                                    1/1

Error: Unable to detect the oneAPI MPI installation at /opt/intel.

error: %prein(intel-mpi-devel-ohpc-2023.1-300.ohpc.7.1.x86_64) scriptlet failed, exit status 1

Error in PREIN scriptlet in rpm package intel-mpi-devel-ohpc
  Verifying        : intel-mpi-devel-ohpc-2023.1-300.ohpc.7.1.x86_64                                                                    1/1

Failed:
  intel-mpi-devel-ohpc-2023.1-300.ohpc.7.1.x86_64

Error: Transaction failed

**Comment by adrianreber** [comment]
This is fixed in 2.7 and will also be fixed in 3.1.
